openedx/core/djangoapps/schedules/signals.py

```diff
@@ -55,17 +59,34 @@ def create_schedule(sender, **kwargs):
 
     upgrade_deadline = _calculate_upgrade_deadline(enrollment.course_id, content_availability_date)
 
+    if course_has_highlights(enrollment.course_id):
+        experience_type = ScheduleExperience.EXPERIENCES.course_updates
+    else:
+        experience_type = ScheduleExperience.EXPERIENCES.default
+
+    if _suppress_schedules_for_some_enrollments(schedule_config):
+        log.debug('Schedules: Enrollment held back from dynamic schedule experiences.')
+        upgrade_deadline_str = None
+        if upgrade_deadline:
+            upgrade_deadline_str = upgrade_deadline.isoformat()
+        analytics.track(
+            'edx.bi.schedule.suppressed',
+            {
+                'user_id': enrollment.user.id,
+                'course_id': unicode(enrollment.course_id),
+                'experience_type': experience_type,
+                'upgrade_deadline': upgrade_deadline_str,
+                'content_availability_date': content_availability_date.isoformat(),
+            }
+        )
```

Comment:
Nit: why not keep this logic encapsulated within the helper function?

Reply:
Given that it's basically just a pass-through of a bunch of context variables from the outer function scope, it felt silly to pass in all of that data into a function which in turn just uses it to call another function.